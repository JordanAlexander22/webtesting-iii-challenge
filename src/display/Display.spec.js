import React from "react";
import * as rtl from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Display from "./Display";

afterEach(rtl.cleanup);
let wrapper;
let Unlocked = () => {
  return wrapper.queryByText("Unlocked");
};
let Open = () => {
  return wrapper.queryByText("Open");
};
let Closed = () => {
  return wrapper.queryByText("Closed");
};
let Locked = () => {
  return wrapper.queryByText("Locked");
};
it("renders without crashing", () => {
  wrapper = rtl.render(<Display />);
  expect(wrapper.container).toMatchSnapshot();
});
it("renders elements properly", () => {
  wrapper = rtl.render(<Display closed={false} locked={false} />);
  expect(wrapper.container).toMatchSnapshot();
  expect(Unlocked()).toBeInTheDocument();
  expect(Unlocked()).toBeVisible();
  expect(Unlocked()).toHaveClass("led", "green-led");
  expect(Open()).toHaveClass("led", "green-led");
});
it("renders elements properly, true", () => {
  wrapper = rtl.render(<Display closed={true} locked={false} />);
  expect(wrapper.container).toMatchSnapshot();
  expect(Unlocked()).toBeInTheDocument();
  expect(Unlocked()).toBeVisible();
  expect(Unlocked()).toHaveClass("led", "green-led");
  expect(Closed()).toHaveClass("led", "red-led");
});
it("renders elements properly, closed and locked", () => {
  wrapper = rtl.render(<Display closed={true} locked={true} />);
  expect(Unlocked()).not.toBeInTheDocument();
  expect(Locked()).toBeInTheDocument();
  expect(Locked()).toBeVisible();
  expect(Locked()).toHaveClass("led", "red-led");
  expect(Closed()).toHaveClass("led", "red-led");
});