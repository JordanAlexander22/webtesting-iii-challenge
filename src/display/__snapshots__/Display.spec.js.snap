// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display should match snapshot 1`] = `
<body>
  <div>
    <div
      class="display panel"
    >
      <div
        class="led green-led"
      >
        Unlocked
      </div>
      <div
        class="led green-led"
      >
        Open
      </div>
    </div>
  </div>
</body>
`;

exports[`renders elements properly 1`] = `
<div>
  <div
    class="display panel"
  >
    <div
      class="led green-led"
    >
      Unlocked
    </div>
    <div
      class="led green-led"
    >
      Open
    </div>
  </div>
</div>
`;

exports[`renders elements properly, true 1`] = `
<div>
  <div
    class="display panel"
  >
    <div
      class="led green-led"
    >
      Unlocked
    </div>
    <div
      class="led red-led"
    >
      Closed
    </div>
  </div>
</div>
`;

exports[`renders without crashing 1`] = `
<div>
  <div
    class="display panel"
  >
    <div
      class="led green-led"
    >
      Unlocked
    </div>
    <div
      class="led green-led"
    >
      Open
    </div>
  </div>
</div>
`;
